src/openrct2/scripting/bindings/ride/ScRide.cpp

```diff
@@ -489,6 +499,29 @@ namespace OpenRCT2::Scripting
         return ::GetRide(_rideId);
     }
 
+    void ScRide::SetBreakdown(const std::string& breakDown)
+    {
+        ThrowIfGameStateNotMutable();
+        auto ride = GetRide();
+        if (ride != nullptr && ride->canBreakDown() && ride->status == RideStatus::open)
+        {
+            auto it = BreakdownMap.find(breakDown);
+            if (it == BreakdownMap.end())
+                return;
+            RidePrepareBreakdown(*ride, it->second);
+        }
+    }
+
+    void ScRide::FixBreakdown()
+    {
+        ThrowIfGameStateNotMutable();
+        auto ride = GetRide();
+        if (ride != nullptr && ride->canBreakDown() && ride->status == RideStatus::open)
```

Comment:
Should we really limit fixing breakdowns to open rides only?

Reply:
I don't think a non-open ride can be broken down at all, want it changed?

Reply:
What if you breakdown the ride, and then close it? Then the breakdown cannot be fixed by plugins?

Reply:
Ah, I'd wrongly remember that you can't close a broken down ride.. but it's double-closing (despawning trains) that's impossible. I'll fix it